packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1522,6 +1567,59 @@ export class TransactionController extends BaseController<
   ): boolean {
     return remoteGasUsed !== localGasUsed;
   }
+
+  private async requestApproval(
+    txMeta: TransactionMeta,
+    { shouldShowRequest } = { shouldShowRequest: true },
+  ) {
+    const id = this.getApprovalId(txMeta);
+    const { origin } = txMeta;
+    const type = ApprovalType.Transaction;
+    const requestData = { txId: txMeta.id };
+
+    try {
+      await this.messagingSystem.call(
+        'ApprovalController:addRequest',
+        {
+          id,
+          origin: origin || ORIGIN_METAMASK,
+          type,
+          requestData,
+        },
+        shouldShowRequest,
+      );
+    } catch (error) {
+      console.info('Failed to request transaction approval', error);
+    }
+  }
+
+  private acceptApproval(txMeta: TransactionMeta) {
+    const id = this.getApprovalId(txMeta);
+
+    try {
+      this.messagingSystem.call('ApprovalController:acceptRequest', id);
+    } catch (error) {
+      console.info('Failed to accept transaction approval request', error);
+    }
+  }
+
+  private rejectApproval(txMeta: TransactionMeta) {
+    const id = this.getApprovalId(txMeta);
+
+    try {
+      this.messagingSystem.call(
+        'ApprovalController:rejectRequest',
+        id,
+        new Error('Rejected'),
```

Comment:
Definitely not blocking, as we can sort this eventually when we don't call `rejectApproval` from within this controller and we do so from the client, but there's a specific `userRejectedRequest` method in the eth-rpc-errors package that returns a proper RPC error for this.